### DISCUSSION TITLE
Недокументированные особенности трека

### DISCUSSION THREAD

**Discussion by fuccsoc** [Chimildic/goofy#117] [Идея] [answered]
Случайно нашел следующие параметры для auido_features: target_anger, target_happiness, target_sadness. Может, стоит добавить информацию о них в документацию?

**Comment by Chimildic** [comment, marked as answer]
Ранее писали, что нашли подобные параметры кажется в коде [майнере треков](http://playlistminer.playlistmachinery.com/). С тех пор [rangeTracks](https://chimildic.github.io/goofy/#/func?id=rangetracks) умеет по ним фильтровать (рассчитывая значения по формулам из майнера). 

О каком источнике в данном случае речь? `target_` в запросе рекомендаций используется, но в официальных доках таких параметров нет. При указании любых имен ключей не будет ошибок. Поэтому до конца не понятна есть ли они на стороне Spotify.

UPD: как отмечается в комментарии, нет объективного подтверждения наличия таких параметров в функции рекомендаций.

**Reply by fuccsoc** [reply]
Речь идёт о getRecommendations из сидов

Проверил оптыным путем, треки реально фильтруются по настроению. Можете проверить сами: при высоком target_sadness треки будут заметно печальнее

**Reply by Chimildic** [reply]
Если посмотреть на вычисляемые значения, то они совсем далеки от заданной цели. И в самих треках не вижу разницы при крайних значениях `0.1` и `0.9`.
```js
  let tracks = Source.getRecomTracks({
    seed_genres: 'indie',
    target_sadness: 0.6
  });

  let cache = getCachedTracks(tracks, { features: {}}).features;
  Object.values(cache).forEach(c => console.log(c.sadness));
```
